src/Command/InfectionCommand.php

```diff
@@ -253,6 +260,38 @@ private function getOutputFormatter(): OutputFormatter
         throw new \InvalidArgumentException('Incorrect formatter. Possible values: "dot", "progress"');
     }
 
+    /**
+     * By default we let a mutant process use as much more memory as an initial test suite consumed.
+     *
+     * @var int
+     */
+    const DEFAULT_MEMORY_LIMIT_MULTIPLIER = 2;
+
+    private function applyMemoryLimitFromPhpUnitProcess(Process $process, PhpUnitAdapter $adapter)
+    {
+        $memoryLimit = $adapter->getMemoryUsed($process->getOutput());
+
+        if ($memoryLimit < 0) {
+            // Cannot detect memory used, not setting any limits
+            return;
+        }
+
+        /*
+         * Since we know how much memory the initial test suite used,
+         * and only if we know, we can enforce a memory limit upon all
+         * mutation processes. Limit is set to be twice the known amount,
+         * because if we know that a normal test suite used X megabytes,
+         * if a mutants uses a lot more, this is a definite error.
+         */
+        $memoryLimit *= self::DEFAULT_MEMORY_LIMIT_MULTIPLIER;
```

Comment:
could you please move this line under condition as we don't use it elsewhere?

Reply:
Done.